app/boards/shields/nibble/nibble.overlay

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <dt-bindings/zmk/matrix-transform.h>
+
+/ {
+	chosen {
+		zmk,kscan = &kscan0;
+		zmk,matrix_transform = &default_transform;
+	};
+
+	kscan0: kscan {
+		compatible = "zmk,kscan-gpio-demux";
+		label = "KSCAN";
+		diode-direction = "col2row";
```

Comment:
Is there really any other option for this other than `col2row`, as this is implemented? Seems like having the other direction would require tweaks to the read code to account for tweaking the demux stuff as well.

If that's the case, perhaps we shouldn't stick w/ the previous naming from the regular matrix KSCAN driver, and just use `input-gpios` and `output-gpios`?

Reply:
Great point. I'd left it generic, but now I'm leaning toward taking the diode direction logic out and only supporting the one direction. As you pointed out, the implementation wouldn't actually work in the reverse direction.